From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 22 Mar 2025 13:34:39 -0500
Subject: [PATCH] Aeryn default config changes

---
 config.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.in b/config.in
index a2a01dda..10d5c3c6 100644
--- a/config.in
+++ b/config.in
@@ -14,7 +14,7 @@ set $down j
 set $up k
 set $right l
 # Your preferred terminal emulator
-set $term foot
+set $term alacritty
 # Your preferred application launcher
 set $menu wmenu-run
 
